digital-deck-server/routes/player.js

```diff
@@ -8,46 +8,68 @@ router.post('/playcard', async function (req, res) {
   var cardIndex = req.body.cardIndex;  // there should only be one of this required, either the card or the index
   var card = req.body.card;           // since the card needs to be added to the table, sending the index of the card in player's hand shouldn't be required
 
+
+  if (isNaN(sessionId) || isNaN(playerId) || isNaN(cardIndex) || isNaN(card)) {
+    res.status(400).send("Invalid call. Needs sessionId, playerId, cardIndex and card as numbers in the query.");
+    return;
+  }
+
+  if (cardIndex < 0 || card < 0 || card > 52) {
+    res.status(400).send("Invalid call. Card Index and card should be in bounds.");
+    return;
+  }
+
   var session = getSession(sessionId);
+  if (!session) {
+    res.status(400).send("Invalid request. Could not find session with Id " + sessionId);
+    return;
+  }
+
   var table = session.table;
   var player = getPlayer(session, playerId);
-
-  try {
-    player.removeCard(cardIndex);
-    table.addCardTop(card);
-    res.status(200).send();
+  if (!player) {
+    res.status(400).send("Invalid request. Could not find player with Id " + playerId + " in session " + sessionId);
+    return;
   }
-  catch (err) {
-    console.log(err);
-    res.status(400).send();
+
+  if(cardIndex > player.cards.length)
+  {
+    res.status(400).send("Invalid request. Can not play cardIndex " + cardIndex + ". Player " + playerId + " only has " + player.cards.length + " cards.");
+    return;
   }
+
+
+  player.removeCard(cardIndex);
+  table.addCardTop(card);
+  res.status(200).send(session);
+
 });
 
 router.post('/drawcard', async function (req, res) {
-    let sessionId = Number(req.body.sessionId);
-    let playerId = Number(req.body.playerId);
-    let numOfCards = Number(req.body.numOfCards);
+  let sessionId = Number(req.body.sessionId);
```

Comment:
You don't want to commit this.  You're undoing the changes I made on my branch.

Reply:
I didn't change anything in this. It is just line number change I think. I don't know why it shows it here

Reply:
It's the spacing. Each line has 2 less space at the beginning. I guess it doesn't really matter.